File: backend/po/PurchaseOrder.java
package com.info5059.casestudy.po;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;
import java.math.BigDecimal;
import java.time.ZonedDateTime;

@Entity
@Data
@RequiredArgsConstructor
public class PurchaseOrder {
    @Id
    @GeneratedValue
    private Long Id;
    private Long vendorid;
    private BigDecimal amount;
    @JsonFormat(pattern = "yyyy-MM-dd@HH:mm:ss")
    private ZonedDateTime datecreated;
    @OneToMany(mappedBy = "poid", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PurchaseOrderLineitem> items = new ArrayList<PurchaseOrderLineitem>();
}

File: frontend/purchaseorder/purchaseorder.ts
import { PurchaseorderItem } from './purchaseorder-item';
/*
    Purcahse Order - interface for product report 
*/
export interface Purchaseorder {
  id: number;
  vendorid: number;
  amount: number;
  items: PurchaseorderItem[];
  datecreated: string;
}
